packages/lending/src/adapter.ts

```diff
@@ -185,6 +185,299 @@ export class Adapter extends common.Web3Toolkit {
     return this.protocolMap[id];
   }
 
+  async openByCollateral(
+    account: string,
+    portfolio: Portfolio,
+    zapToken: common.Token,
+    zapAmount: string,
+    collateralToken: common.Token,
+    collateralAmount: string,
+    debtToken: common.Token,
+    slippage = defaultSlippage
+  ): Promise<OperationOutput> {
+    const output: OperationOutput = {
+      destAmount: '0',
+      afterPortfolio: portfolio.clone(),
+      logics: [],
+    };
+    try {
+      // 1. ---------- validate params ----------
+      if (collateralAmount === '0') throw new OperationError('collateralAmount', 'ZERO_AMOUNT');
+
+      const srcCollateral = portfolio.findSupply(collateralToken);
+      if (!srcCollateral) throw new OperationError('collateralToken', 'UNSUPPORTED_TOKEN');
+
+      const destBorrow = portfolio.findBorrow(debtToken);
+      if (!destBorrow) throw new OperationError('debtToken', 'UNSUPPORTED_TOKEN');
+
+      const supplyAmountDelta = new common.TokenAmount(collateralToken.wrapped, collateralAmount).sub(
+        srcCollateral.balance
+      );
+      if (!srcCollateral.validateSupplyCap(supplyAmountDelta.amount)) {
+        throw new OperationError('collateralAmount', 'SUPPLY_CAP_EXCEEDED');
+      }
+
+      // 2. ---------- zap swap ----------
+      let supplyInput = new common.TokenAmount(collateralToken.wrapped, '0');
+      if (zapAmount !== '0' && !zapToken.wrapped.is(collateralToken.wrapped)) {
+        const swapper = this.findSwapper([zapToken, collateralToken]);
+        let swapQuotation: SwapperQuoteFields;
+        try {
+          swapQuotation = await swapper.quote({
+            input: new common.TokenAmount(zapToken, zapAmount),
+            tokenOut: collateralToken.wrapped,
+            slippage,
+          });
+        } catch {
+          throw new OperationError('zapAmount', 'NO_ROUTE_FOUND_OR_PRICE_IMPACT_TOO_HIGH');
+        }
+        const swapTokenLogic = swapper.newSwapTokenLogic(swapQuotation);
+        output.logics.push(swapTokenLogic);
+        // 2-1. the supply amount is the swap quotation output
+        supplyInput = swapQuotation.output;
+        output.afterPortfolio.supply(supplyInput.token, supplyInput.amount);
+      } else if (zapToken.wrapped.is(collateralToken.wrapped)) {
```

Comment:
Could you please review whether we can replace `else if` with `else` since `zapAmount` is not being checked in this context? The absence of an `else` statement, when not dealing with error handling, seems like a potential error imo.

Moreover, then we can move `output.afterPortfolio.supply(supplyInput.token, supplyInput.amount);` outside of `if-else` block.